SConstruct

```diff
@@ -179,18 +177,43 @@ for k in platform_opts.keys():
     for o in opt_list:
         opts.Add(o)
 
-for x in module_list:
-    module_enabled = True
-    tmppath = "./modules/" + x
-    sys.path.insert(0, tmppath)
+# Detect modules.
+opts.Add("custom_modules", "Directory path containing custom modules to build along with built-in modules.", "")
+
+modules_detected = {}
+module_search_paths = ["modules"]  # Built-in path.
+
+if ARGUMENTS.get("custom_modules"):
+    paths = ARGUMENTS.get("custom_modules").split(os.pathsep)
```

Comment:
Wouldn't it be better to use `,`, which should be usable cross-platform? It makes both documentation and usage simpler if users don't need to differentiate between Windows and POSIX.

Reply:
If that makes it cross-platform, sure, but needs a mention that it's a comma-separated list of paths explicitly then, as you suggested below.